WordPress/Sniffs/WP/CapabilitiesSniff.php

```diff
@@ -0,0 +1,301 @@
+		'edit_css'                    => true,
+		'assign_categories'           => true,
+		'assign_post_tags'            => true,
+	);
+
+	/**
+	 * List of deprecated core capabilities.
+	 *
+	 * User Levels were  deprecated in version 3.0.
+	 * To be updated after every major release.
+	 * Last updated for WordPress 4.9.6.
+	 *
+	 * @link https://github.com/WordPress/wordpress-develop/blob/master/tests/phpunit/tests/user/capabilities.php
+	 *
+	 * @since 1.0.0
+	 *
+	 * @var array All deprecated capabilities in core.
+	 */
+	protected $deprecated_capabilities = array(
+		'level_10' => true,
+		'level_9'  => true,
+		'level_8'  => true,
+		'level_7'  => true,
+		'level_6'  => true,
+		'level_5'  => true,
+		'level_4'  => true,
+		'level_3'  => true,
+		'level_2'  => true,
+		'level_1'  => true,
+		'level_0'  => true,
+	);
+
+	/**
+	 * Process the parameters of a matched function.
+	 *
+	 * @since 1.0.0
+	 *
+	 * @param int    $stackPtr        The position of the current token in the stack.
+	 * @param array  $group_name      The name of the group which was matched.
+	 * @param string $matched_content The token content (function name) which was matched.
+	 * @param array  $parameters      Array with information about the parameters.
+	 *
+	 * @return void
+	 */
+	public function process_parameters( $stackPtr, $group_name, $matched_content, $parameters ) {
+
+		$position = $this->target_functions[ $matched_content ];
+		if ( ! isset( $parameters[ $position ] ) ) {
+			return;
+		}
+
+		$next_not_empty = $this->phpcsFile->findNext(
+			Tokens::$emptyTokens,
+			$parameters[ $position ]['start'],
+			$parameters[ $position ]['end'] + 1,
+			true
+		);
+
+		$matched_parameter = $this->strip_quotes( $this->tokens[ $next_not_empty ]['content'] );
+		if ( isset( $this->core_capabilities[ $matched_parameter ] ) ) {
+			return;
+		}
+
+		$custom_capabilities = $this->merge_custom_array( $this->custom_capabilities, array(), true );
+		if ( isset( $custom_capabilities[ $matched_parameter ] ) ) {
+			return;
+		}
+
+		if ( isset( $this->deprecated_capabilities[ $matched_parameter ] ) ) {
+			$this->phpcsFile->addError(
+				'The capatibility "%s" found in function call "%s()" has been deprecated since WordPress version 3.0.',
+				$next_not_empty,
+				'DeprecatedCapability',
+				array(
+					$matched_parameter,
+					$matched_content,
+				)
+			);
+			return;
+		}
+
+		if ( isset( $this->core_roles[ $matched_parameter ] ) ) {
+			$this->phpcsFile->addError(
+				'Capabilities should be used instead of roles. Found "%s" in function call "%s()"',
+				$next_not_empty,
+				'RoleFound',
+				array(
+					$matched_parameter,
+					$matched_content,
+				)
+			);
+			return;
+		}
+
+		if ( false === $this->check_only_known_caps ) {
+			$this->phpcsFile->addWarning(
+				'"%s" is an unknown role or capability. Check the "%s()" function call to ensure it is a capability and not a role.',
+				$next_not_empty,
+				'UnknownCapabilityFound',
```

Comment:
Not a blocker, but the error code could just be `Unknown` as `Capability` is already in the sniff name, so the full error code would be `WordPress.WP.Capability.Unknown`.